htlcswitch/link.go

```diff
@@ -1223,6 +1219,25 @@ func (l *channelLink) processHtlcResolution(resolution invoices.HtlcResolution,
 	return nil
 }
 
+// getResolutionFailure returns the wire message that a htlc resolution should
+// be failed with.
+func getResolutionFailure(resolution invoices.HtlcResolution,
+	amount lnwire.MilliSatoshi) lnwire.FailureMessage {
+
+	// If the resolution has been resolved as part of a MPP timeout, we need
+	// to fail the htlc with lnwire.FailMppTimeout.
+	if resolution.Outcome == invoices.ResultMPPTimeout {
+		return &lnwire.FailMPPTimeout{}
```

Comment:
Below we use a constructor. Don't know if there are already constructors for every failure type.

Reply:
We tend to not have them for failures with no fields, `FailFinalExpiryTooSoon` has a constructor but it's the exception among the no field failures. The majority is no fields -> no constructor so I'm going to leave as is.

Reply:
Ok